project/MimaExcludes.scala

```diff
@@ -55,7 +55,11 @@ object MimaExcludes {
       ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.sql.catalog.Catalog.getFunction"),
       ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.sql.catalog.Catalog.databaseExists"),
       ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.sql.catalog.Catalog.tableExists"),
-      ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.sql.catalog.Catalog.functionExists")
+      ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.sql.catalog.Catalog.functionExists"),
+      // [SPARK-17816] Fix ConcurrentModificationException issue in BlockStatusesAccumulator
+      ProblemFilters.exclude[DirectAbstractMethodProblem]("org.apache.spark.util.AccumulatorV2.merge"),
```

Comment:
Hm, but this class didn't have any change in its API, not at this point. I can't see why it would still be needed. In fact there's a merge conflict here anyway, can you just try removing it and see if it passes?

Looking strong to me otherwise, great cleanup.